### PR TITLE
Target branch selection settings

### PR DESCRIPTION
**Pull request by lekhmanrus**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build related changes
- [ ] Documentation
- [ ] Other: 

## What is the current behavior?
Branch selection is based on a local branch only.

Issue Number: #53

## What is the new behavior?
Adds Target branch selection settings to allow control over which default branch for publish would be selected:
- selection based on the local branch
- reading from the `.gitreview` file

Users can setup/override the default behaviour.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/GitExtensions.GerritPlugin/FormGerritPublish.cs [hunk 1]
```diff
@@ -137,18 +139,24 @@ private bool PublishChange(IWin32Window owner)
...
                 branchName = string.Join("/", branchParts.Skip(2));
 
                 // Don't use the Gerrit change number as a topic branch.
-
                 if (int.TryParse(branchName, out _))
                 {
-                    branchName = null;
+                    return null;
+                }
+                else
```
Comment (RussKie) [inline src/GitExtensions.GerritPlugin/FormGerritPublish.cs:152]: [nit] `else` is not required here.

**Review by RussKie — COMMENTED** [review]
👍

File: src/GitExtensions.GerritPlugin/FormGerritPublish.cs (hunk 1 above)
Comment (lekhmanrus) [reply src/GitExtensions.GerritPlugin/FormGerritPublish.cs]: Thanks!)

**Comment by lekhmanrus** [comment]
Could somebody review this pull request, please?

**Review by gerhardol — COMMENTED** [review]
Seem OK to me in a brief review. Not a user for this plugin.
@lekhmanrus Do you want an approval to get going, you will have to handle any issues regardless?

**Comment by lekhmanrus** [comment]
@gerhardol yes, please

**Review by gerhardol — APPROVED** [review]
approving as requested

**Pull request merged**